### PR TITLE
NT-85 Add commit and rollback hooks to SQLObject

### PR DESCRIPTION
**Pull request by stahlscott**
- Adding new signal hooks to SQLObject to publish events on Commit / Rollback to help fix the data integrity issue between bazman and engage

### CODE REVIEW

File: sqlobject/tests/test_transactions.py [hunk 1]
```diff
@@ -28,6 +46,11 @@ def test_transaction():
         b = TestSOTrans.byName('bob', connection=trans)
         b.name = 'robert'
         trans.commit()
+
+        assert len(watcher.log) == 1
+        assert watcher.log[0][0][0][0] == 'TestSOTrans'
```
Comment (aqualad) [inline sqlobject/tests/test_transactions.py]: lol

Comment (aqualad) [reply sqlobject/tests/test_transactions.py]: very progressive

File: sqlobject/dbconnection.py [hunk 2]
```diff
@@ -827,6 +831,11 @@ def rollback(self):
                     inst.expire()
         self._makeObsolete()
 
+    def _send_event(self, signal):
+        sub_caches = [(sub[0], sub[1].allIDs()) for sub in self.cache.allSubCachesByClassNames().items()]
```
Comment (timmartin19) [inline sqlobject/dbconnection.py]: What's the `sub` object?  Also, this is a good case for tuple unpacking

`[x, y.allIDs(), for x, y in self.cache.allSubCachesByClassNames().items()]`

Obviously better names would be preferred.

Comment (stahlscott) [reply sqlobject/dbconnection.py]: Good point; I used chr for chr what they used a few lines up from here. Will refactor for clarity

**Pull request merged**